tests/phpunit/tests/speculative-loading/wpGetSpeculationRules.php

```diff
@@ -0,0 +1,378 @@
+<?php
+/**
+ * Tests for the wp_get_speculation_rules() function.
+ *
+ * @package WordPress
+ * @subpackage Speculative Loading
+ */
+
+/**
+ * @group speculative-loading
+ * @covers ::wp_get_speculation_rules
+ */
+class Tests_Speculative_Loading_wpGetSpeculationRules extends WP_UnitTestCase {
+
+	private $prefetch_config  = array(
+		'mode'      => 'prefetch',
+		'eagerness' => 'conservative',
+	);
+	private $prerender_config = array(
+		'mode'      => 'prerender',
+		'eagerness' => 'conservative',
+	);
+
+	public function set_up() {
+		parent::set_up();
+
+		add_filter(
+			'template_directory_uri',
+			static function () {
+				return content_url( 'themes/template' );
+			}
+		);
+
+		add_filter(
+			'stylesheet_directory_uri',
+			static function () {
+				return content_url( 'themes/stylesheet' );
+			}
+		);
+
+		update_option( 'permalink_structure', '/%year%/%monthnum%/%day%/%postname%/' );
+	}
+
+	/**
+	 * Tests speculation rules output with prefetch for the different eagerness levels.
+	 *
+	 * @ticket 62503
```

Comment:
All of these repeating `@ticket 62503` annotations seem excessive/extraneous. `@ticket` annotations are mostly used to run a set of tests while working on a ticket AFAIK. We can already run all of the tests for this ticket using `--group= speculative-loading` so we can remove all of the `@ticket` annotations entirely, right?

Reply:
I don't think they exist temporarily. They're useful for context, makes it easy to find out as part of which ticket a test was added. Of course right now `@ticket 62503` is synonymous with `@group speculative-loading`, but this will no longer be the case in the near future, as soon as any enhancement or bug is fixed via another ticket.

Reply:
I always understood `@ticket` as something to be used when adding regression tests for _bugs_, not enhancements. Also not a fan of adding `@ticket` everywhere for a new feature like this. But I know some people do that, so 🤷